modules/sd_models.py

```diff
@@ -227,20 +229,43 @@ def load_model_weights(model, checkpoint_info, vae_file="auto"):
     sd_vae.load_vae(model, vae_file)
 
 
+def get_config(checkpoint_info):
+    path = checkpoint_info[0]
+    model_config = checkpoint_info.config
+    try:
+        checkpoint = torch.load(path)
+        c_dict = checkpoint["state_dict"]
+        v2_key = "model.diffusion_model.input_blocks.2.1.transformer_blocks.0.attn2.to_k.weight"

```

Comment:
Looks like this key is present in all models. 

This is only in V2 models from what I can see:

```
v2_key = "cond_stage_model.model.ln_final.weight"
```

Reply:
Good call. Used your key - I'll also add this to my conversion scripts for dreambooth, as that's where this came from.

Also added a bit to ensure we don't load the checkpoint twice if calling from reload_model_weights, a check in the whole checkpoint for the key if the model was created without a "state_dict" entry, and made the log message a bit more useful.

Tested with the v2.0 768-v checkpoint, v2.0 512 checkpoint, several v1x checkpoints, and the new v2.1 checkpoints. All load as intended with no yaml files needed.

Reply:
How do you differenciate between the 512 and 768 models with the updated code? I dont see anything special dealing with that.

Reply:
It's the number of training steps. It's really the only difference between them, and it's the hack Huggingface suggested to use...

Need to find the number of steps for the 512 v2.1 model - but only if the 2.1 512 doesn't use v-prediction.